### PR TITLE
Add redis migrator, for moving all keys from one redis to another

### PR DESCRIPTION
**Pull request by mcasper**

### CODE REVIEW

**Comment by ajvb** [comment]
+1

File: redis/redis_migrator.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Original Author: Matt Casper (@mcasper/mattc@procore.com)
+
+require 'redis'
+
+# These can be new Redis servers, these can be different namespaces on the same
+# Redis server, whatever. Replace these with the appropriate Redis connections.
+# All keys from `old_redis` will be moved with the appropriate values to
+# `new_redis`.
+old_redis = Redis.new
```
Comment (jfrost) [inline redis/redis_migrator.rb]: Probably the nicest way to handle these would be to have command line args for the redis URL?

File: redis/redis_migrator.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        new_redis.lpush(key, value)
+      end
+    when "set"
+      members = old_redis.smembers(key)
+      members.each do |member|
+        new_redis.sadd(key, member)
+      end
+    when "zset"
+      # This ranges over all the keys in the zset, making sure to get back
+      # the scores as well. They scores/values are returned in reverse order
```
Comment (mcasper) [inline redis/redis_migrator.rb]: `They` typo, should be `The`

File: redis/redis_migrator.rb (hunk 1 above)
Comment (mcasper) [inline redis/redis_migrator.rb]: But if they need to connect using a namespace, they're going to have to pull in another gem (which should maybe be included here by default) and specifically use Ruby classes to get the namespace.

Comment (jfrost) [inline redis/redis_migrator.rb]: Optional namespace args as well?

Pull in the gem conditionally?

Comment (mcasper) [inline redis/redis_migrator.rb]: Hmm I suppose so.

**Comment by mcasper** [comment]
Ok @jfrost, added the argument parsing bits.

Passed all testing locally, Thursday will be the final test

**Comment by mcasper** [comment]
It works!

**Pull request merged**